rero_ils/modules/patron_transaction_events/serializers/csv.py

```diff
@@ -0,0 +1,190 @@
+    """Serialize patron transaction event search for csv."""
+
+    def serialize_search(self, pid_fetcher, search_result, links=None,
+                         item_links_factory=None):
+        """Serialize a search result.
+
+        :param pid_fetcher: Persistent identifier fetcher.
+        :param search_result: Elasticsearch search result.
+        :param links: Dictionary of links to add to response.
+        :param item_links_factory: Factory function for record links.
+        """
+
+        def generate_csv():
+            headers = dict.fromkeys(self.csv_included_fields)
+
+            # write the CSV output in memory
+            line = Line()
+            writer = csv.DictWriter(
+                line, quoting=csv.QUOTE_ALL, fieldnames=headers)
+            writer.writeheader()
+            yield line.read()
+
+            for hit in search_result:
+                event = hit.to_dict()
+                parent = self.get_resource(
+                    PatronTransaction,
+                    event.get('parent', {}).get('pid')
+                )
+                transaction_date = parse_datetime(event.get('creation_date'))
+
+                # Load related resources used to fill the file.
+                csv_data = {
+                    'category': event.get('category'),
+                    'type': event.get('type'),
+                    'subtype': event.get('subtype'),
+                    'amount': round(event.get('amount'), 2),
+                    'transaction_date': transaction_date.isoformat()
+                }
+
+                if pid := event.get('patron', {}).get('pid'):
+                    record = self.get_resource(Patron, pid)
+                    csv_data |= _extract_patron_data(record)
+                if pid := event.get('patron_type', {}).get('pid'):
+                    record = self.get_resource(PatronTypesSearch(), pid)
+                    csv_data |= _extract_patron_type_data(record)
+                if pid := event.get('operator', {}).get('pid'):
+                    record = self.get_resource(Patron, pid)
+                    csv_data |= _extract_operator_data(record)
+                if pid := event.get('library', {}).get('pid'):
+                    record = self.get_resource(Library, pid)
+                    csv_data |= _extract_transaction_library_data(record)
+                    csv_data['transaction_date'] = transaction_date\
+                        .astimezone(tz=record.get_timezone())\
+                        .isoformat()
+                if pid := parent.loan_pid:
+                    loan = self.get_resource(Loan, pid)
+                    document = self.get_resource(Document, loan.document_pid)
+                    item = self.get_resource(Item, loan.item_pid)
+                    csv_data |= _extract_document_data(document)
+                    csv_data |= _extract_item_data(item, self)
+
+                # write csv data
+                data = self.process_dict(csv_data)
```

Comment:
many dict comprehension is used to clean data in `_extract_xxx` methods (see comments below).
you could clean data in one time here before `process_dict()`.